frontend/src/components/DataSelector/DatasetSelector.js

```diff
@@ -27,19 +45,85 @@ function generateJSXOptionsWithDatasetNames(datasetNames) {
  */
 function DatasetSelector() {
     const {setSelectedDatasetName, setSelectedDataModelName1, setSelectedDataModelName2} = useDataSelectorContext();
-
     const [datasetNames, setDatasetNames] = useState([]);
-    const [isOptionSelected, setIsOptionSelected] = useState(false);
+    const [isDisabledSelect, setIsDisabledSelect] = useState(false);
     const [datasetSelectorError, setDatasetSelectorError] = useState(null);
 
-    const handleSelectChange = useCallback((event) => {
-        setSelectedDatasetName(event.target.value);
-        setIsOptionSelected(true);
+    const handleOptionChange = useCallback((event) => {
+        setSelectedDatasetName(event.value);
 
         // Clear previous data model names
         setSelectedDataModelName1(null);
         setSelectedDataModelName2(null);
-    }, [setSelectedDatasetName, setSelectedDataModelName1, setSelectedDataModelName2, setIsOptionSelected]);
+    }, [setSelectedDatasetName, setSelectedDataModelName1, setSelectedDataModelName2]);
+
+    /**
+     * Custom styles for the Select component.
+     *
+     * @type {Object} Custom style for Select component.
+     */
+    const customStyles = {
+        // Styles for the control (main container)
+        control: (base) => ({
+            ...base,
+            backgroundColor: isDisabledSelect ? color.disable.controlBackground : color.enable.controlBackground,
+            borderRadius: 0,
+            border: 0,
+            boxShadow: "none"
+        }),
+        // Styles for the options
+        option: (base) => ({
+            ...base,
+            backgroundColor: isDisabledSelect ? color.disable.optionBackground : color.enable.optionBackground,
+            color: isDisabledSelect ? color.disable.optionText : color.enable.optionText,
+            '&:hover': {
+                backgroundColor: isDisabledSelect ? color.disable.optionHoverBackground : color.enable.optionHoverBackground
+            }
+        }),
+        // Styles for the menu
+        menu: (base) => ({
+            ...base,
+            backgroundColor: isDisabledSelect ? color.disable.menuBackground : color.enable.menuBackground
+        }),
+        // Styles for the selected option
+        singleValue: (base) => ({
+            ...base,
+            color: isDisabledSelect ? color.disable.singleValueText : color.enable.singleValueText
+        }),
+        // Styles for the search input
+        input: (base) => ({
+            ...base,
+            color: isDisabledSelect ? color.disable.inputText : color.enable.inputText
+        })
+    };
+
+    /**
+     * Custom them for the Select component.
+     *
+     * @param theme Custom them.
+     * @returns {*&{colors: string}} Custom colors.
+     */
+    const customTheme = useCallback((theme) => {
+        return (
+            {
+                ...theme,
+                colors: ""
+            }
+        );
```

Comment:
```suggestion
        return ({ ...theme, colors: "" });
```

ça me choque pas de l'avoir sur une ligne, l'objet est suffisamment petit pour ne pas gêner la lecture